geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/operations/PutAllRequestOperationHandler.java

```diff
@@ -52,29 +57,64 @@
           "Region passed does not exist: " + regionName);
     }
 
+    ThreadState threadState = null;
+    SecurityService securityService = messageExecutionContext.getCache().getSecurityService();
+    boolean perKeyAuthorization = false;
+    if (messageExecutionContext
+        .getConnectionStateProcessor() instanceof ProtobufConnectionAuthorizingStateProcessor) {
+      threadState = ((ProtobufConnectionAuthorizingStateProcessor) messageExecutionContext
+          .getConnectionStateProcessor()).prepareThreadForAuthorization();
+      // Check if authorized for entire region
```

Comment:
Except for the resource permission operation, this looks the same as the one for get-all. Could this be refactored into a common base class?

Reply:
The permissions and operation being called are different.  There's probably a creative refactoring that can be done here, but I'd rather wait until after storage team fixes their putAll and getAll operations before we start abstracting out pieces.